### PR TITLE
Fix UP032 auto-fix with integers

### PR DESCRIPTION
**Pull request by JonathanPlasse**
- Related to #4523

### CODE REVIEW

File: crates/ruff/src/rules/pyupgrade/snapshots/ruff__rules__pyupgrade__tests__UP032_2.py.snap [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+source: crates/ruff/src/rules/pyupgrade/mod.rs
+---
+UP032_2.py:1:1: UP032 [*] Use f-string instead of `format` call
+  |
+1 | "{.real}".format(1)
+  | ^^^^^^^^^^^^^^^^^^^ UP032
+2 | 
+3 | "{0.real}".format(1)
```
[the remark this answers is not in the document]
Comment (charliermarsh) [reply crates/ruff/src/rules/pyupgrade/snapshots/ruff__rules__pyupgrade__tests__UP032_2.py.snap:9]: Great, thank you 🙏

[the remark this answers is not in the document]
Comment (charliermarsh) [reply crates/ruff/src/rules/pyupgrade/snapshots/ruff__rules__pyupgrade__tests__UP032_2.py.snap:9]: It looks like these work too?

```py
>>> f"{0b101.real}"
'5'
>>> f"{0o101.real}"
'65'
```

File: crates/ruff/resources/test/fixtures/pyupgrade/UP032_2.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+"{.real}".format(1)
+
+"{0.real}".format(1)
+
+"{a.real}".format(a=1)
```
Comment (charliermarsh) [inline crates/ruff/resources/test/fixtures/pyupgrade/UP032_2.py:5]: Can you add some more test cases here, for floats, octal integers, complex, etc.?

File: crates/ruff/src/rules/pyupgrade/snapshots/ruff__rules__pyupgrade__tests__UP032_2.py.snap (hunk 1 above)
[the remark this answers is not in the document]
Comment (JonathanPlasse) [reply crates/ruff/src/rules/pyupgrade/snapshots/ruff__rules__pyupgrade__tests__UP032_2.py.snap:9]: Indeed

**Pull request merged**